iceoryx_posh/include/iceoryx_posh/popo/listener.hpp

```diff
@@ -27,13 +27,50 @@
 #include "iceoryx_posh/popo/notification_attorney.hpp"
 #include "iceoryx_posh/popo/notification_callback.hpp"
 #include "iceoryx_posh/popo/trigger_handle.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
```

Comment:
~~Why is this include needed?~~ Should be moved to `.inl`

Reply:
I disagree. The `inl` file is the header implementation and adding there headers hides this from the user which is including this header file.
Every header which is required to use this should be listed here and not secretly hidden in the `inl` file.

Now we need a third one to decide the tie.

Mysterious stranger please add a thumbs up to one of the remarks to break the tie.

Reply:
.hpp and .inl work differently from .hpp and .cpp insofar that the user is aware of all dependencies (as the compiler has to be). We should probably not make it harder for the user to find them, but on the other hand consider.

If we think of them as all pasted in one file (which is what the preprocessor does), where would we include the headers? While one can  include the headers directly before use (at the latest moment), it is highly unusual. I am not aware that it will cause any errors, for the compiler it should not make any difference. But it makes it harder for the user to understand the dependencies.

I am fine with either, but would like to avoid implementation specific headers in the file the user includes. This does not really lead to nice structure (IMO) with .inl files.

In the end I dislike the .inl file approach and prefer the detail/internal namespace approach as it also reduces namespace pollution with implementation details. Then we can have an implementation header using the detail sub namespace. This header would now include the implementation specific headers. If one is to imitate this with .inl I think we would include the implementation specific header in the .inl file.

TLDR: if we want to mimic what is usually done with detail namespaces we would include the header in the .inl file. However, it is not as clean as having the detail header and detail sub namespace.